src/NServiceBus.Metrics.AcceptanceTests/Tests/QueueLength/When_sending_message.cs

```diff
@@ -38,26 +37,43 @@ public class When_sending_message : NServiceBusAcceptanceTest
                 }))
                 .WithEndpoint<Receiver1>()
                 .WithEndpoint<Receiver2>()
-                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2)
+                .WithEndpoint<MonitoringSpy>()
+                .Done(c => c.Headers1.Count == 2 && c.Headers2.Count == 2 && SequencesReported(c))
                 .Run()
                 .ConfigureAwait(false);
+        }
 
-            var sessionIds = new [] { AssertHeaders(context.Headers1), AssertHeaders(context.Headers2)};
+        static bool SequencesReported(Context context)
+        {
+            try
+            {
+                var sessionIds = new[]
+                {
+                    AssertHeaders(context.Headers1),
+                    AssertHeaders(context.Headers2)
+                };
 
-            var data = JObject.Parse(context.Data);
-            var counters = (JArray)data["Counters"];
-            var counterTokens = counters.Where(c => c.Value<string>("Name").StartsWith("Sent sequence for"));
+                var data = JObject.Parse(context.Data);
+                var counters = (JArray) data["Counters"];
+                var counterTokens = counters.Where(c => c.Value<string>("Name").StartsWith("Sent sequence for"));
 
-            foreach (var counter in counterTokens)
-            {
-                var tags = counter["Tags"].ToObject<string[]>();
-                var counterBasedKey = tags.GetTagValue("key");
-                var type = tags.GetTagValue("type");
+                foreach (var counter in counterTokens)
+                {
+                    var tags = counter["Tags"].ToObject<string[]>();
+                    var counterBasedKey = tags.GetTagValue("key");
+                    var type = tags.GetTagValue("type");
 
-                CollectionAssert.Contains(sessionIds, counterBasedKey);
-                Assert.AreEqual(2, counter.Value<int>("Count"));
-                Assert.AreEqual("queue-length.sent", type);
+                    CollectionAssert.Contains(sessionIds, counterBasedKey);
+                    Assert.AreEqual(2, counter.Value<int>("Count"));
+                    Assert.AreEqual("queue-length.sent", type);
+                }
+            }
+            catch
+            {
+                return false;
```

Comment:
Because we want to check the asserts until they are true or test execution time elapses. If we would throw here the test would end in failure potentially prematurely.

The general problem here is that we don't have a reasonable way of saying when is the time report should include sequence data.